lib/src/xdg_screen_cast_portal.dart

```diff
@@ -0,0 +1,248 @@
+  final int nodeId;
+
+  /// Opaque identifier.
+  final String id;
+
+  final int x;
+  final int y;
+  final int width;
+  final int height;
+
+  /// The type of the content which is being screen casted.
+  final ScreenCastAvailableSourceType sourceType;
+
+  ScreenCastStream({
+    required this.nodeId,
+    required this.id,
+    required this.x,
+    required this.y,
+    required this.width,
+    required this.height,
+    required this.sourceType,
+  });
+  @override
+  int get hashCode => Object.hash(nodeId, id, x, y, width, height, sourceType);
+
+  @override
+  bool operator ==(other) {
+    if (identical(this, other)) return true;
+
+    return other is ScreenCastStream &&
+        other.nodeId == nodeId &&
+        other.id == id &&
+        other.x == x &&
+        other.y == y &&
+        other.width == width &&
+        other.height == height &&
+        other.sourceType == sourceType;
+  }
+
+  @override
+  String toString() =>
+      '$runtimeType(node ID: $nodeId, id: $id, position: ($x, $y), size: ($width, $height), sourceType: $sourceType})';
+}
+
+/// Screen cast portal.
+class XdgScreenCastPortal {
+  final DBusRemoteObject _object;
+  final String Function() _generateToken;
+
+  DBusObjectPath? _sessionPath;
+
+  XdgScreenCastPortal(this._object, this._generateToken);
+
+  /// Get the version of this portal.
+  Future<int> getVersion() => _object
+      .getProperty('org.freedesktop.portal.ScreenCast', 'version',
+          signature: DBusSignature('u'))
+      .then((v) => v.asUint32());
+
+  /// Get the available screen cast source types.
+  Future<Set<ScreenCastAvailableSourceType>> getAvailableSourceTypes() =>
+      _object
+          .getProperty(
+              'org.freedesktop.portal.ScreenCast', 'AvailableSourceTypes',
+              signature: DBusSignature('u'))
+          .then((v) {
+        final value = v.asUint32();
+        final types = <ScreenCastAvailableSourceType>{};
+        if (value & 1 != 0) types.add(ScreenCastAvailableSourceType.monitor);
+        if (value & 2 != 0) types.add(ScreenCastAvailableSourceType.window);
+        if (value & 4 != 0) types.add(ScreenCastAvailableSourceType.virtual);
+        return types;
+      });
+
+  /// Get the available screen cast cursor modes.
+  Future<Set<ScreenCastAvailableCursorMode>> getAvailableCursorModes() =>
+      _object
+          .getProperty(
+              'org.freedesktop.portal.ScreenCast', 'AvailableCursorModes',
+              signature: DBusSignature('u'))
+          .then((v) {
+        final value = v.asUint32();
+        final types = <ScreenCastAvailableCursorMode>{};
+        if (value & 1 != 0) types.add(ScreenCastAvailableCursorMode.hidden);
+        if (value & 2 != 0) types.add(ScreenCastAvailableCursorMode.embedded);
+        if (value & 4 != 0) types.add(ScreenCastAvailableCursorMode.metadata);
+        return types;
+      });
+
+  /// Create a screen cast session.
+  Future<List<ScreenCastStream>> createSession({
```

Comment:
This should return `Stream<ScreenCastSesion>` - that way the session can be closed by the API user. See the location portal for example. I was going to make a PR to this branch that fixes this, but I haven't had the time to get to it. If you're not sure exactly what I mean when I get the time I'll land this PR as is, and then immediately follow up with the commit that changes this.